pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -206,6 +209,13 @@ func (p *Pod) isUnretriableGroup() bool {
 	return false
 }
 
+func (p *Pod) getClock() clock.Clock {
```

Comment:
Instead a function like this - can we please make sure the clock is always set, this will make the code much simpler to reason about, and can prevent panics when someone assumes it is set and forgets to call the function. It is very hard to see this type of a bug when reviewing and seeing on GH only fragment of the code.

Reply:
@mimowo So, pod doesnot have a constructor so, we can no more implement options in it. What other approach do you think would fit in this situation ?

Reply:
I synced with @PBundyra and he should be able to help more, but I think it you could make this work with:
- make the `lastActiveTime` and `sortInactivePods` to be receiving the (p *Pod) received from `FindMatchingWorkloads`
- then we could have a function like `NewPod` accepting the clock (possibly as an option). In production code pass the real clock, and in tests the fake one

You can follow the pattern for the WorkloadReconciler from workload_controller.go